client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -244,13 +244,7 @@ class Commit(
    *  @return Failure if this commit or other has not yet been saved, or if they do not belong to the same Repository; the merged commit otherwise.
    */
   def merge(other: Commit, message: Option[String] = None)(implicit ec: ExecutionContext) = {
-    if (!saved)
-      Failure(new IllegalCommitSavedStateException("This commit must be saved"))
-    else if (!other.saved)
-      Failure(new IllegalCommitSavedStateException("Other commit must be saved"))
-    else if (other.repo.id != repo.id)
-      Failure(new IllegalArgumentException("Two commits must belong to the same repository"))
-    else
+    checkCommit("This commit must be saved", Some(other), Some("Other commit must be saved")).flatMap(_ => {
```

Comment:
```suggestion
    checkCommit("This commit must be saved").flatMap(_ => other.checkCommit("Other commit must be saved")).flatMap(_ => {
```
seems more idiomatic, right? No need to worry about arbitrary number of arguments.

Reply:
Yeah it does, let me fix that.

Reply:
Done.